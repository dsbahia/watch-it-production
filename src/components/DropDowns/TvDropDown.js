import React from "react";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";

function TvDropDown() {
  return (
    <DropdownButton id="item-tv-dropdown" title="Tv Shows">
      <Dropdown.Item data-testid="top-rated-link" href="/top-rated-tv-shows">
        Top Rated
      </Dropdown.Item>
      <Dropdown.Item data-testid="airing-link" href="/airing-tv-shows">
        Airing
      </Dropdown.Item>
    </DropdownButton>
  );
}

export default TvDropDown;
