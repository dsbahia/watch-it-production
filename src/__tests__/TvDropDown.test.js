import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TvDropDown from "../components/DropDowns/TvDropDown";

describe("TvDropdown", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<TvDropDown />);
    expect(asFragment).toMatchSnapshot();
  });

  it("displays 'Tv Shows' title", () => {
    const { getByText } = render(<TvDropDown />);
    expect(getByText("Tv Shows")).toBeInTheDocument();
  });

  it("displays dropdowns with correct paths", () => {
    const { getByTestId, getByText } = render(<TvDropDown />);
    const dropdownButton = getByText("Tv Shows");

    fireEvent.click(dropdownButton);

    const topRated = getByTestId("top-rated-link");
    const airing = getByTestId("airing-link");

    expect(topRated).toHaveAttribute("href", "/top-rated-tv-shows");
    expect(airing).toHaveAttribute("href", "/airing-tv-shows");
  });
});
