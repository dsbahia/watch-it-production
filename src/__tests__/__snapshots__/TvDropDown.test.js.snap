// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvDropdown renders correctly 1`] = `[Function]`;
